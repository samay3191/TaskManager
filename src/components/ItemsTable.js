import React from "react";
import { useDispatch } from "react-redux";
import { removeItem, selectItem } from "../actions/action";
import CheckedIcon from "../images/icon-checked.svg";
import DeleteIcon from "../images/icon-delete.svg";

const Item = ({ data }) => {
  const dispatch = useDispatch();
  const { text, selected } = data;
  
  return (
    <div className="flex p-2 bg-white justify-between rounded-md my-1">
      <div>{text}</div>
      <div className="flex text-blue-600">
        {!selected && (
          <div className="p-1 hover:bg-blue-100 rounded-full">
            <img
              src={CheckedIcon}
              alt="select"
              className="w-4 align-middle mx-1 cursor-pointer"
              onClick={() => dispatch(selectItem(data.text))}
            />
          </div>
        )}
        <div className="p-1 hover:bg-red-100 rounded-full">
          <img
            src={DeleteIcon}
            alt="delet"
            className="w-4 align-middle cursor-pointer"
            onClick={() => dispatch(removeItem(data.text))}
          />
        </div>
      </div>
    </div>
  );
};

const ItemsTable = ({ data }) => {
  return (
    <div className="flex flex-col-reverse">
      {data.map((item) => (
        <Item key={item.text} data={item} />
      ))}
    </div>
  );
};

export default ItemsTable;
