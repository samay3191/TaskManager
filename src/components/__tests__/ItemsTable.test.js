import React from "react";
import { shallow } from "enzyme";
import ItemsTable from "../ItemsTable";

const initialProps = {
  data: [
    {
      id: "abc",
      text: "Food",
      selected: false,
    },
  ],
};

let component;

describe("<ItemsTable />", () => {
  beforeAll(() => {
    component = shallow(<ItemsTable {...initialProps} />);
  });

  it("should render ItemsTable component", () => {
    expect(component).toMatchSnapshot();
  });
});
