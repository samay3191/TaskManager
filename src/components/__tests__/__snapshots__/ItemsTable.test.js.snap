// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemsTable /> should render ItemsTable component 1`] = `
<div
  className="flex flex-col-reverse"
>
  <Item
    data={
      Object {
        "id": "abc",
        "selected": false,
        "text": "Food",
      }
    }
    key="abc"
  />
</div>
`;
